src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -136,7 +136,7 @@ public String run(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNu
      * @param containerId The container ID.
      */
     public void stop(@Nonnull EnvVars launchEnv, @Nonnull String containerId) throws IOException, InterruptedException {
-        LaunchResult result = launch(launchEnv, false, "stop", containerId);
+        LaunchResult result = launch(launchEnv, false, "stop", "--time=1", containerId);
```

Comment:
But the current implementation will always run into the timeout (cat won't terminate). So as long as the containers are not able to stop gracefully it's one second which could be saved.